### PR TITLE
send custom slf4j logs as events in amqp exchange

### PR DESCRIPTION
**Pull request by benrejebmoh**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

feature : convert slf4j logs to rabbitmq events with binding to calculation task

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change?** *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: src/main/resources/application.yml [hunk 1]
```diff
@@ -8,14 +14,8 @@ spring:
...
-          binder: rabbit
-          destination: core-valid-run-task
-          group: job-launcher
-      rabbit:
-        default:
-          consumer:
-            auto-bind-dlq: true
-            max-attempts: 1
+          destination: cse-d2cc-run-task
+          group: cse-d2cc-job-launcher
```
Comment (TheMaskedTurtle) [inline src/main/resources/application.yml:18]: Why did you change that ?

File: src/main/resources/application.yml [hunk 2]
```diff
@@ -1,4 +1,10 @@
+logback:
+  amqp:
+    exchange: "task-manager-logs-exchange"
```
Comment (TheMaskedTurtle) [inline src/main/resources/application.yml:3]: I think the "" are not required here

File: src/main/java/com/farao_community/farao/gridcapa/job_launcher/JobLauncherService.java [hunk 3]
```diff
@@ -14,7 +14,7 @@
  */
 @Service
 public class JobLauncherService {
-    private static final Logger LOGGER = LoggerFactory.getLogger(JobLauncherService.class);
+    private static final Logger BUSINESS_LOGGER = LoggerFactory.getLogger("BUSINESS_LOGGER");
```
Comment (TheMaskedTurtle) [inline src/main/java/com/farao_community/farao/gridcapa/job_launcher/JobLauncherService.java:17]: Isn't it possible to define this logger as a bean for the whole application ? Maybe not useful in job-launcher case but it might be useful later

**Review by TheMaskedTurtle — CHANGES_REQUESTED** [review]

File: src/main/java/com/farao_community/farao/gridcapa/job_launcher/JobLauncherService.java (hunk 3 above)
Comment (benrejebmoh) [reply src/main/java/com/farao_community/farao/gridcapa/job_launcher/JobLauncherService.java:17]: I agree with the idea, to extract that aside, but not using a bean. 
I would use a class that could be instantiated only if given a task id, that we use to set it as extra field in MDC.
What do you think?

File: src/main/resources/application.yml (hunk 1 above)
Comment (benrejebmoh) [reply src/main/resources/application.yml:18]: with comparing to the deployment config i assumed that it is not useful .. i guess forgotten after changing function by consumer?
For the destination and group i didn't gave it too much interest as it have to be reconfigured any way in dev mode. Also overridden in prod mode..

Comment (TheMaskedTurtle) [reply src/main/resources/application.yml:18]: Yes it is not overriden in the prod files but it is required in the build of the application otherwise it wouldn't be taken into account. We don't have to redefine these parameters everytime in the prod deployments because there are defined here. So there are required. Maybe not for the "binder: rabbit" we can test but the other ones are not the default values for rabbit binder so we need them.

File: src/main/java/com/farao_community/farao/gridcapa/job_launcher/JobLauncherService.java (hunk 3 above)
Comment (TheMaskedTurtle) [reply src/main/java/com/farao_community/farao/gridcapa/job_launcher/JobLauncherService.java:17]: ok, no risk if we instantiate this class several times ? In different classes for example ?

File: src/main/resources/application.yml [hunk 4]
```diff
@@ -8,14 +14,13 @@ spring:
     stream:
       bindings:
         run-task:
-          binder: rabbit
```
Comment (TheMaskedTurtle) [inline src/main/resources/application.yml:11]: You are sure it is good without this line ?

File: src/test/resources/application.yml [hunk 5]
```diff
@@ -1,2 +1,6 @@
+logback:
+  amqp:
+    exchange: "task-manager-logs-exchange"
```
Comment (TheMaskedTurtle) [inline src/test/resources/application.yml:3]: you can remove the ""

**Review by TheMaskedTurtle — APPROVED** [review]
